src/stactools/modis/metadata.py

```diff
@@ -157,11 +157,15 @@ def get_exception(xpath: str) -> Exception:
                 "ParameterName", missing_element("ParameterName")
             ).replace(" ", "_")
             band_qa_percent_cloud_cover = measured_parameter.find_text(
-                "QAPercentCloudCover"
+                "QAStats/QAPercentCloudCover"
             )
-            if qa_percent_cloud_cover:
+            if band_qa_percent_cloud_cover:
                 qa_percent_cloud_cover[name] = int(band_qa_percent_cloud_cover)
 
+        if qa_percent_cloud_cover and not qa_percent_not_produced_cloud:
+            assert len(set(qa_percent_cloud_cover.values())) == 1
```

Comment:
Can we add an assertion message, so if/when this blows up we know why right away? Suggestion might need some wordsmithing.

```suggestion
            assert len(set(qa_percent_cloud_cover.values())) == 1, "more than one qa_percent_cloud_cover exists in this item, which is not supported at this time"
```

Reply:
Added a message.